mkosi/run.py

```diff
@@ -381,7 +367,10 @@ def chroot_cmd(root: Path, *, options: Sequence[PathString] = (), network: bool
     else:
         cmdline += ["--unshare-net"]
 
-    return cmdline
+    # No exec here because we need to clean up the /work directory afterwards.
+    cmdline += ["sh", "-c", f"$0 \"$@\" && rm -rf {root / 'work'} || exit $?"]
+
+    return apivfs_cmd(root) + cmdline
```

Comment:
So this pastes together something like `bwrap [options] sh -c "stuff" sh -c "stuff"`? I'm surprised that works without anything to combine the two.

Reply:
The latter one just becomes an argument to the earlier one. Works great, allows you to do arbitrary setup without losing the interface where you just string command lines together.

Reply:
From the bash man page:

```
       -c        If the -c option is present, then commands are read from the first non-option argument command_string.  If there are arguments after the command_string, the first argument is assigned to $0 and any remaining argu‐
                 ments are assigned to the positional parameters.  The assignment to $0 sets the name of the shell, which is used in warning and error messages.
```

Reply:
Makes sense, neat! A little comment explaining this would be good, though, since this does rely on what `apivfs_cmd` is to be of the form `sh -c "$0 \"$@\""` then.